### PR TITLE
Add set/get for textclock attributes

### PR DESCRIPTION
**Pull request by otommod**
It's not very pretty the way it's implemented but hey, it works.  My
reason for wanting this was to allow "declarative" construction of
clocks.

If you'd prefer it to be more "class-like", I'd be happy to change it.

### CODE REVIEW

**Comment by actionless** [comment]
looks fine but it's missing docstrings

**Comment by otommod** [comment]
I added them.  Where are the docs build though?

**Comment by actionless** [comment]
default make target will build the docs

File: lib/wibox/widget/textclock.lua [hunk 1]
```diff
@@ -55,6 +55,39 @@ function textclock.new(format, timeout, timezone)
...
+        timezone = tzid and TimeZone.new(tzid) or TimeZone.new_local()
+        w:force_update()
+    end
+    function w.get_timezone()
+        return tzid
+    end
+
+    --- Set the clock's timeout
+    -- @tparam number How often the clock is updated
+    function w.set_timeout(_, new_timeout)
```
Comment (Elv13) [inline lib/wibox/widget/textclock.lua]: Maybe call it refresh rate?

File: lib/wibox/widget/textclock.lua [hunk 2]
```diff
@@ -55,6 +55,39 @@ function textclock.new(format, timeout, timezone)
         w._timer:again()
         return true -- Continue the timer
     end
+
+    --- Set the clock's format
+    -- @tparam string format The new time format.  This can contain pango
+    -- markup
+    function w.set_format(_, new_format)
```
Comment (Elv13) [inline lib/wibox/widget/textclock.lua]: Please do not add functions in the constructor, add them outside and store the attributes in `self._private`

Also, use `@property` for properties and add an extra empty line under the documentation block.

File: lib/wibox/widget/textclock.lua (hunk 1 above)
Comment (otommod) [reply lib/wibox/widget/textclock.lua]: Sure.

File: lib/wibox/widget/textclock.lua (hunk 2 above)
Comment (otommod) [reply lib/wibox/widget/textclock.lua]: I thought as much.  Fair enough.  However, how should I then add the methods?  I mean, right now we're just returning a regular ol' textbox with a timer attached, not a textclock "class".

Comment (psychon) [reply lib/wibox/widget/textclock.lua]: Elv13's method seems to be:
```lua
local foo = {}
function foo:set_format(new_format)
    self._private.format = new_format
end

function constructor()
    local w = textbox()
    gears.table.crush(w, foo)
    return w
end
```

**Comment by otommod** [comment]
Updated.

**Comment by Elv13** [comment]
Thanks!

**Pull request merged**